File: backend/app/schemas/agent.py
"""
Agent Schemas
Pydantic models for agent API
"""

from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime


class ChatRequest(BaseModel):
    """Request to chat with an agent"""
    message: str = Field(description="User's message")
    agent_id: str = Field(default="service_management", description="Agent to use")
    session_id: Optional[str] = Field(default=None, description="Session ID for conversation continuity")
    stream: bool = Field(default=False, description="Enable streaming response")

    class Config:
        json_schema_extra = {
            "example": {
                "message": "Camera service có đang chạy không?",
                "agent_id": "service_management",
                "stream": False
            }
        }


class ChatResponse(BaseModel):
    """Response from agent"""
    response: str = Field(description="Agent's response")
    agent_id: str = Field(description="Agent that generated response")
    session_id: str = Field(description="Session ID")
    tool_calls: Optional[List[Dict[str, Any]]] = Field(default=None, description="Tools used by agent")
    timestamp: str = Field(description="Response timestamp")

    class Config:
        json_schema_extra = {
            "example": {
                "response": "Service đang chạy và đã kết nối WebSocket ✅",
                "agent_id": "service_management",
                "session_id": "session_123",
                "tool_calls": [
                    {
                        "tool": "check_service_status",
                        "args": {"service_name": "camera_management"}
                    }
                ],
                "timestamp": "2026-01-11T10:30:00"
            }
        }


class AgentInfo(BaseModel):
    """Information about an agent"""
    agent_id: str = Field(description="Agent identifier")
    class_name: str = Field(description="Python class name")
    description: str = Field(description="Agent description")

    class Config:
        json_schema_extra = {
            "example": {
                "agent_id": "service_management",
                "class_name": "ServiceManagementAgent",
                "description": "Agent chuyên quản lý services"
            }
        }


class AgentHealthResponse(BaseModel):
    """Agent system health status"""
    status: str = Field(description="Health status")
    agents_registered: int = Field(description="Number of registered agents")
    openai_configured: bool = Field(description="Whether OpenAI API is configured")
    message: str = Field(description="Status message")

File: frontend-ts/src/types/agent.ts
/**
 * AI Agent Types
 */

export interface SuggestedAction {
  label: string;
  message: string;
}

export interface Message {
  id: string;
  role: 'user' | 'assistant' | 'system';
  content: string;
  timestamp: Date;
  toolCalls?: ToolCall[];
  status?: 'pending' | 'success' | 'error';
  suggestedActions?: SuggestedAction[];
}

export interface ToolCall {
  tool: string;
  args: Record<string, any>;
  id?: string;
}

export interface ChatRequest {
  message: string;
  agent_id?: string;
  session_id?: string;
  stream?: boolean;
}

export interface ChatResponse {
  response: string;
  agent_id: string;
  session_id: string;
  tool_calls?: ToolCall[];
  timestamp: string;
}

export interface AgentInfo {
  agent_id: string;
  class_name: string;
  description: string;
}

export interface ServiceStatus {
  service_name: string;
  is_running: boolean;
  websocket_connected: boolean;
  status: 'healthy' | 'degraded' | 'stopped';
  message: string;
  pid?: number;
  cpu_percent?: number;
  memory_mb?: number;
}

export interface QuickAction {
  label: string;
  icon: string;
  message: string;
  condition?: () => boolean;
}
